### PR TITLE
Drop DataPlaneService ConfigMaps and Secrets

### PR DESCRIPTION
**Pull request by fao89**
- Drops the ConfigMaps and Secrets fields from OpenStackDataPlaneService.
- Updates all services under config/services to use the OpenStackDataPlaneService.Spec.DataSources field instead.
- Update docs for the switch to DataSources.

Copied: https://github.com/openstack-k8s-operators/dataplane-operator/pull/923

### CODE REVIEW

File: docs/assemblies/custom_resources.adoc [hunk 1]
```diff
@@ -15397,7 +15397,7 @@ AutoscalingSection defines the desired state of the autoscaling service
 
 | enabled
 | Enabled - Whether OpenStack autoscaling service should be deployed and managed
-| bool
+| *bool
```
Comment (jpodivin) [inline docs/assemblies/custom_resources.adoc:15400]: These changes to the docs appear completely unrelated to the PR.

**Review by jpodivin — CHANGES_REQUESTED** [review]
I don't agree with dropping of the labels, at very least not all of them. There is still need for fine grained resource selection and I don't think removing labels supporting that is especially useful.

File: docs/assemblies/custom_resources.adoc (hunk 1 above)
Comment (fao89) [reply docs/assemblies/custom_resources.adoc:15400]: make docs wasn't added into pre-commit hook, I added it here: https://github.com/openstack-k8s-operators/openstack-operator/pull/851

**Comment by fao89** [comment]
> I don't agree with dropping of the labels, at very least not all of them. There is still need for fine grained resource selection and I don't think removing labels supporting that is especially useful.

I just copied @slagle PRs, but I believe the labels were removed because it didn't have a pattern among the services.
I think we can add labels using the webhook, this way we would ensure a pattern

**Review by rabi — COMMENTED** [review]
/lgtm

**Comment by rabi** [comment]
Needs a rebase after the docs patch it seems.
